mm2src/adex_cli/src/activation_scheme_db/activation_scheme_impl.rs

```diff
@@ -0,0 +1,70 @@
+use anyhow::{anyhow, bail, Result};
+use log::{debug, error};
+use serde_json::Value as Json;
+use std::collections::HashMap;
+
+use super::init_activation_scheme::get_activation_scheme_path;
+use crate::helpers::read_json_file;
+use crate::logging::{error_anyhow, error_bail};
+
+#[derive(Default)]
+pub(crate) struct ActivationScheme {
+    scheme: HashMap<String, Json>,
+}
+
+impl ActivationScheme {
+    pub(crate) fn get_activation_method(&self, coin: &str) -> Option<&Json> { self.scheme.get(coin) }
+
+    fn init(&mut self) -> Result<()> {
+        let mut scheme_source: Vec<Json> = Self::load_json_file()?;
+        self.scheme = scheme_source
+            .iter_mut()
+            .filter_map(Self::get_coin_activation_command)
+            .collect();
+        Ok(())
+    }
+
+    fn get_coin_activation_command(element: &mut Json) -> Option<(String, Json)> {
+        Self::get_coin_activation_command_impl(element).ok()
+    }
+
+    fn get_coin_activation_command_impl(element: &mut Json) -> Result<(String, Json)> {
+        let coin = element
+            .get_mut("coin")
+            .ok_or_else(|| error_anyhow!("Failed to get coin pair, no coin value"))?
+            .as_str()
+            .ok_or_else(|| error_anyhow!("Failed to get coin pair, coin is not str"))?
+            .to_string();
+        let mut command = element
+            .get_mut("command")
```

Comment:
Since this is not built in this release and unused other than dev purposes at the moment, consider this non-blocking. However, I would suggest this command is checked against a list of valid commands that might be here. The JSON is downloaded from a remote server. If you have write access to this JSON, you can change the commands within it, and these will run instead while doing `./adex-cli enable <COIN>`
eg, editing KMD value to `show_priv_key` instead:
```
./adex-cli enable KMD
Enabling asset: KMD
Failed to deserialize response from data: "{\"result\":{\"coin\":\"KMD\",\"priv_key\":\"<MY PRIVATE KEY>\"}}", error: invalid type: map, expected a string at line 1 column 10
Failed to send: `enable`, error:
```

It's not critical, but a check to disallow such behavior is trivial to add.

Reply:
cc @rozhkovdmitrii

Reply:
@Alrighttt,
Thank you for remark!
I suggested a little more comprehensive decision in the [PR](https://github.com/KomodoPlatform/komodo-defi-framework/pull/1910/files) that introduces a ActivationRequest types. I guess it helps it prevent malicious intentions that could happen.